python/pyspark/pandas/series.py

```diff
@@ -373,8 +373,6 @@ class Series(Frame, IndexOpsMixin, Generic[T]):
     ----------
     data : array-like, dict, or scalar value, pandas Series
         Contains data stored in Series
-        If data is a dict, argument order is maintained for Python 3.6
-        and later.
         Note that if `data` is a pandas Series, other arguments should not be used.
     index : array-like or Index (1d)
         Values must be hashable and have the same length as `data`.
```

Comment:
We can also remove https://github.com/apache/spark/blob/33f19012dcdab1e4c880816b1377b2ad881bf1ae/python/pyspark/pandas/series.py#L347-L359